### ISSUE TITLE
"Server not found" when using FQDN via reverse proxy

### ISSUE THREAD

**Issue by ri3ch99e** [jellyfin/jellyfin-vue#2105]
### Description of the bug

**TL;DR**
**Works:** Access jellyfin-vue directly with _http://IP:Port_ and enter jellyfin server as _http://IP:Port_
**Doesn't work:** Access jellyfin-vue via fqdn (_https://jellyfin.home_) and enter jellyfin server as either _http://IP:Port_ or _https://jellyfinsvr.home_

**Story mode:**
Navigating to the jellyfin-vue container, jellyfin-vue prompts for the Jellyfin server address. The jellyfin-vue server is hosted on _http://<local_ip>:8097_ and Jellyfin on _http://<local_ip>:8096_. They are provisioned with docker-compose and are in the same jellyfin_default network. Debian server with firewalld disabled.

If I enter the IP:Port into jellyfin-vue when accessing directly via http, everything works fine.
But if I access jellyfin-vue through the proxy (e.g. _https://jellyfin.home_ which directs to _http://<local ip>:8097_) I get the “Server Not Found” error. Ditto if I use _https://jellyfinsvr.home_ which is reverse proxied to _http://<local_ip>:8096_ using the same ssl certificate.

### Steps to reproduce

1. Navigate to https://jellyfin.home (a reverse proxy for http://<local_ip>:8097)
2. Prompted for Jellyfin server address
3. Enter address (either http://<local_ip>:8096 or https://jellyfinsvr.home)
4. Server not found.

### Expected behavior

jellyfin-vue discovers Jellyfin server

### Logs

```text
None that I could find.
```

### Platform

Linux

### Browser

Firefox

### Jellyfin server version

"latest"

### Additional context

```text
version: '3'
services:
  jellyfin:
    image: docker.io/jellyfin/jellyfin:latest
    container_name: jellyfin
    ports: 
      - "8096:8096"
    volumes:
      - ./config:/config
      - ./cache:/cache
      - /mnt/biggie/media:/media:ro
    group_add:
      - "105"
      - "44"
      - "226"
      - "128"
    devices:
      - /dev/dri/renderD128:/dev/dri/renderD128
      - /dev/dri/card0:/dev/dri/card0
    environment:
      - ROC_ENABLE_PRE_VEGA=1
      - JELLYFIN_PublishedServerUrl=https://jellyfinsvr.home
      - PUID=<docker user id>
      - GUID=<docker group id>

  jellyfin-vue:
    image: ghcr.io/jellyfin/jellyfin-vue:unstable
    container_name: jellyfin-vue
    ports:
      - "8097:80"
    environment:
      - DEFAULT_SERVERS=<local_ip>:8096
```

**Comment by ri3ch99e** [comment]
New data point: using FQDN http://jellyfin.home does allow me to use http://<local_ip>:8096 as the server, but not http://jellyfinsvr.home. Maybe SSL validation issue?

**Comment by ri3ch99e** [comment]
Okay, looks like an SSL Certificate validation issue on the client side. Using a clean browser (ie. private browsing on a new computer), I was able to get it working by FIRST going to Jellyfin, accepting the certificate THEN going to Vue and entering in the Jellyfin https server.

Connecting Vue to a Jellyfin server really aught to be done server-side and not client-side (at least, from my limited knowledge I assume this is the case). I shouldn't have to send out instructional information just for folks in my household to use Vue... nor should I have to use CA validated certificates for something that exists only in a homelab environment that isn't exposed to the Internet.

**Comment by ferferga** [comment]
@ri3ch99e This is not something we can't fix, it's browsers

If you want to use Vue without an https certificate, you need to deploy your own version working under http

https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content
